inst/template/mode.R.template

```diff
@@ -55,16 +55,29 @@
     },
 
     n_state = function() {
-      mode_{{name}}_n_state(private$ptr_)
+      mode_{{name}}_n_output(private$ptr_)
     },
 
     n_state_full = function() {
-      mode_{{name}}_n_state_full(private$ptr_)
+      mode_{{name}}_n_output_full(private$ptr_)
     },
 
     update_state = function(pars = NULL, time = NULL,
                             state = NULL, index = NULL,
                             set_initial_state = NULL, reset_step_size = NULL) {
+      dim_state <- dim(state)
```

Comment:
Here just dropping extra values/columns if a vector or matrix corresponds to the full output size, but leaving all other validation to the c++ layer as before. So also not sure whether this splitting of validation logic makes sense, but it keeps the changes here to a minimum.